app/packages/core/src/utils/hooks/useQueryState.tsx

```diff
@@ -52,12 +57,7 @@ export const useQueryState = <S extends QueryState = QueryState>(initialState?:
     // parameter contains a valid time, so that we are really using the "last 15 minutes" and not the times from the
     // `timeEnd` and `timeStart` parameters.
     if (initialQueryFromUrl.current === true) {
-      if (
-        'time' in parsedQueryString &&
-        typeof parsedQueryString.time === 'string' &&
-        times.includes(parsedQueryString.time) &&
-        parsedQueryString.time !== 'custom'
-      ) {
+      if (queryStringHasTime(parsedQueryString)) {
```

Comment:
Where is this function used again, that we can not leave the statement inline?

Reply:
This function isn't about re-usablity, but about it being a [`type-guard`](https://www.typescriptlang.org/docs/handbook/advanced-types.html#user-defined-type-guards).

I improved the types inside `times.ts` by narrowing down the type of `timeOptions`. Instead of it being a `Recod<string, ...>`, I made it a `Record<TTimeQuick, ...>`. This is more precise and should be preferred. 

The `Record<string,...>` version wouldn't give us type-errors when timeOptions is accessed with a string like so:
```ts
// type of option is { label: string; seconds: number; }, but at runtime it's actually undefined
const option = timeOptions["unknownProperty"]
```